### DISCUSSION TITLE
onvif device test tool test behavior issue.

### DISCUSSION THREAD

**Discussion by snew310** [onvif/testspecs#75] [Q&A] [answered]
Hi Expert,

When I try to use device test tool to test us product. I always get fail from test item MEDIA-2-1-9-v14.12 JPEG VIDEO ENCODER CONFIGURATION. Do you know why this item MUST use bit rate 64000 to test the product ? Because our product just support 384 to 32768. The bit rate 64000 is over our bit rate range already. So, could you please help to clarify this problem ? 
Thanks.

Joshua

**Comment by madhurao68** [comment, marked as answer]
Hello,

There is a requirement in ONVIF Media Service Specification, Section 5.5.5 SetVideoEncoderConfiguration:
"A device shall adapt an out of range BitrateLimit instead of returning a fault."

So, the client may send any values and this is also applicable to the ONVIF Device Test Tool. Device shall not return the fault but adapt BitrateLimit to the supported one.

**Comment by snew310** [comment]
Hi expert,

Thank you for your reply. It's very helpful and clear for me.

Joshua
